dcmgr/lib/dcmgr/cli/network.rb

```diff
@@ -834,16 +834,16 @@ def show(uuid=nil)
 Network UUID:
   <%= nw.canonical_uuid %>
 Dynamic IP Address Range:
-<%- unless nw.ipv4_u32_dynamic_range_array.empty? -%>
-  <%= IPAddress::IPv4::parse_u32(nw.ipv4_u32_dynamic_range_array.shift) %> - <%= IPAddress::IPv4::parse_u32(nw.ipv4_u32_dynamic_range_array.last) %>
-<%- end -%>
+<%- nw.dhcp_range_dataset.each { |r| -%>
+  <%= IPAddress::IPv4::parse_u32(r.range_begin.to_u32) %> - <%= IPAddress::IPv4::parse_u32(r.range_end.to_u32) %>
```

Comment:
You are converting an IPv4 classes to u32 and then immediately converting them back to IPv4. Use the IPv4 classes as is. It would also be good to explicitly convert them to strings.

```
  <%= r.range_begin.to_s %> - <%= r.range_end.to_s %>
```